src/planscape/planning/tests/test_views.py

```diff
@@ -221,6 +252,164 @@ def test_delete_multiple_planning_areas(self):
         self.assertEqual(PlanningArea.objects.count(), 1)
 
 
+class UpdatePlanningAreaTest(TransactionTestCase):
+    def setUp(self):
+        self.user = User.objects.create(username='testuser')
+        self.user.set_password('12345')
+        self.user.save()
+        self.geometry = {'type': 'MultiPolygon',
+                         'coordinates': [[[[1, 2], [2, 3], [3, 4], [1, 2]]]]}
+        storable_geometry = GEOSGeometry(json.dumps(self.geometry))
+        self.old_name = 'Westley'
+        self.old_notes = "I know something you don't know."
+        self.planning_area = _create_planning_area(self.user, self.old_name, storable_geometry, self.old_notes)
+
+        self.user2 = User.objects.create(username='testuser2')
+        self.user2.set_password('12345')
+        self.user2.save()
+        self.planning_area2 = _create_planning_area(self.user2, 'test plan2', storable_geometry, self.old_notes)
+
+        self.new_name = 'Inigo'
+        self.new_notes = 'I am not left handed.'
+        
+    def test_update_notes_and_name(self):
+        self.client.force_login(self.user)
+        response = self.client.post(
+            reverse('planning:update_planning_area'),
+            {'id': self.planning_area.pk,
+             'name': self.new_name,
+             'notes': self.new_notes},
+            content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content, json.dumps(
+            {"id": self.planning_area.pk}).encode())
+        planning_area = PlanningArea.objects.get(pk=self.planning_area.pk)
+        self.assertEqual(planning_area.name, self.new_name)
+        self.assertEqual(planning_area.notes, self.new_notes)
+
+    def test_update_notes_only(self):
+        self.client.force_login(self.user)
+        response = self.client.post(
+            reverse('planning:update_planning_area'),
+            {'id': self.planning_area.pk,
+             'notes': self.new_notes},
+            content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.content, json.dumps(
+            {"id": self.planning_area.pk}).encode())
+        planning_area = PlanningArea.objects.get(pk=self.planning_area.pk)
+        self.assertEqual(planning_area.name, self.old_name)
+        self.assertEqual(planning_area.notes, self.new_notes)
+
+    def test_update_name_only(self):
```

Comment:
Just a future note: If we're letting users update the names for their plans and want plan names to be unique (for each user), then we'll need to return some kind of failure message that's different from a generic 'update failed'

Reply:
Name uniqueness is actually enforced by the unique constraint, but I'll add a todo later to add more tests showing the uniqueness enforcement.